typings/index.d.ts

```diff
@@ -162,24 +162,22 @@ declare namespace commander {
      *
      *     // optional argument
      *     program.option('-c, --cheese [type]', 'add cheese [marble]');
-     *
-     * @param {string} flags
-     * @param {string} [description]
-     * @param {((arg1: any, arg2: any) => void) | RegExp} [fn] function or default
-     * @param {*} [defaultValue]
-     * @returns {Command} for chaining
      */
-    option(flags: string, description?: string, fn?: ((arg1: any, arg2: any) => void) | RegExp, defaultValue?: any): Command;
-    option(flags: string, description?: string, defaultValue?: any): Command;
+    option(flags: string, description: string, regexp: RegExp, defaultValue?: string): Command;
```

Comment:
1) A boolean flag can take a default value, which can be a string _or a boolean_.
2) Where the overload order does not matter, I prefer the more common case first.

So I suggest: (add boolean and reorder):
```js
     option(flags: string, description?: string, defaultValue?: string | boolean): Command;
     option<T>(flags: string, description: string, fn: (value: string, previous: T) => T, defaultValue?: T): Command;
     option(flags: string, description: string, regexp: RegExp, defaultValue?: string): Command;
```

Reply:
Okey, nothing to add ?

ready to merge ?